httpserver/src/main/kotlin/tbdex/sdk/httpserver/handlers/SubmitOrder.kt

```diff
@@ -0,0 +1,86 @@
+package tbdex.sdk.httpserver.handlers
+
+import io.ktor.http.*
+import io.ktor.server.application.*
+import io.ktor.server.request.*
+import io.ktor.server.response.*
+import tbdex.sdk.httpclient.models.ErrorDetail
+import tbdex.sdk.httpserver.models.CallbackError
+import tbdex.sdk.httpserver.models.ErrorResponse
+import tbdex.sdk.httpserver.models.ExchangesApi
+import tbdex.sdk.httpserver.models.SubmitCallback
+import tbdex.sdk.protocol.models.Message
+import tbdex.sdk.protocol.models.MessageKind
+import tbdex.sdk.protocol.models.Order
+import tbdex.sdk.protocol.models.Quote
+
+/**
+ * Handles the submission of an order by parsing the incoming message, validating the submission,
+ * and executing the specified callback if provided.
+ *
+ * @param call The [ApplicationCall] instance representing the HTTP call.
+ * @param exchangesApi The [ExchangesApi] instance for interacting with TBDex exchanges.
+ * @param callback The optional callback function to be executed after successful order submission.
+ */
+@Suppress("TooGenericExceptionCaught", "MaxLineLength")
+suspend fun submitOrder(
```

Comment:
Curious, are there any affordances for the same order being submitted twice?

Reply:
Haha that is indeed an issue: https://github.com/orgs/TBD54566975/projects/29/views/4?pane=issue&itemId=44223554

So if the previous message in the exchange is an order, we should get an error because `validNext` is either `close` or `orderStatus`. However we also need to implement this: https://github.com/orgs/TBD54566975/projects/29/views/4?pane=issue&itemId=44223555

to lock the exchange from two orders being added simultaneously. (Really multiple messages in general, the issue is not specific to order messages)